### PR TITLE
Add Che Server functionality for devfile v1 overrides

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Add Che Server functionality for devfile v1 overrides.
Screenshot from 2021-04-19 17-55-07

### What issues does this PR fix or reference?
fixed https://github.com/eclipse/che/issues/19424

### CODE REVIEW

File: src/containers/FactoryLoader.tsx [hunk 1]
```diff
@@ -84,21 +84,10 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
...
-        overrideKeys.push(regExpExecArray[0]);
-      }
-      if (overrideKeys.length > 0) {
-        let currentVal = this.overrideDevfileObject;
-        for (let index = 1; index < overrideKeys.length; index++) {
-          currentVal[overrideKeys[index]] = index === overrideKeys.length - 1 ? val : {};
-          currentVal = currentVal[overrideKeys[index]];
-        }
-      }
+    if (!key.startsWith('override.')) {
```
Comment (sleshchenko) [inline src/containers/FactoryLoader.tsx:87]: Do you think it makes sense to keep this private func after you modified it to be really tiny? or better to inline if into https://github.com/eclipse-che/che-dashboard/blob/CHE-19424/src/containers/FactoryLoader.tsx#L220

I prefer a bit the second option

**Review by sleshchenko — APPROVED** [review]
I haven't tested yet but changes LGTM

**Review by akurinnoy — APPROVED** [review]

File: src/containers/FactoryLoader.tsx (hunk 1 above)
Comment (olexii4) [reply src/containers/FactoryLoader.tsx]: I have fixed it

Comment (olexii4) [reply src/containers/FactoryLoader.tsx]: https://github.com/eclipse-che/che-dashboard/pull/219/files#diff-4a8e21344754b4ee47f503546e5d899120cc756884f5b1c3b266399ddc3ec003R217

**Pull request merged**